### ISSUE TITLE
Restrict the use of passPerPreset

### ISSUE THREAD

**Issue by taion** [babel/babel#6921]
cc @loganfsmyth 

Feature change request

### Input Code
N/A

### Babel/Babylon Configuration (.babelrc, package.json, cli command)
Anything with `passPerPreset`

### Expected Behavior
The use of `passPerPreset` should be restricted

### Current Behavior
`passPerPreset` is allowed in user configurations

### Possible Solution
Disallow using `passPerPreset` in user Babel configurations.

### Context
The majority of current usages of `passPerPreset` in user code originally date to a misunderstanding about plugin ordering with Relay. Relay examples now no longer use `passPerPreset`: https://github.com/relayjs/relay-starter-kit/pull/102, and I've opened an issue with the Apollo devs to remove derived usages there as well: https://github.com/apollographql/starwars-server/pull/5.

Correspondingly, there may not be any real use cases for `passPerPreset` in user code any more. As such, given that this feature has historically not been used very often, and has led to odd interactions between existing presets due to just using a rare code path (e.g. https://github.com/babel/babel/issues/4162), I think it would make sense to heavily limit its use.

Even in cases where it does arise, it's not the correct scope – the scope instead looks more like "this preset or this plugin should have its own pass", rather than "every preset or plugin in this config (or every plugin in this preset) should have its own pass".

### Your Environment
N/A

**Comment by xtuc** [comment]
I agree that we should restrict its usage. After a quick search it seems that some users are relying on it [GitHub search](https://github.com/search?utf8=%E2%9C%93&q=passPerPreset+filename%3A.babelrc&type=Code) even if it's not documented.

**Comment by taion** [comment]
You can also see that substantially all those cases are GraphQL-related cases, owing ultimately to that initial Relay thing.

I think @loganfsmyth mentioned that babel-minify still needed it? Ultimately I think `passPerPreset` is sort of the wrong concept – like I said above, I can see there being a valuable idea in "this preset runs in its own pass", but ultimately I don't think this option should be exposed to end-users like this.

**Comment by hzoo** [comment]
We said we should at least only allow it in a preset (like minify) then end-users wouldn't be able to set it themselves.

**Comment by taion** [comment]
Even for minify, one goal would be to run the entire minify preset after all the earlier presets, no? I feel like there are a couple of different concepts here that are a little bit mixed up:
- "Run this preset in its own pass"
- "Run each part of this preset in its own pass"
